src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private MyLinkedList.Node<T> head;
+    private MyLinkedList.Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> last = tail;
+        final Node<T> newNode = new Node<>(last, value, null);
+        tail = newNode;
+        if (last == null) {
+            head = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        MyLinkedList.Node<T> oldNode = findNodeByIndex(index);
+        MyLinkedList.Node<T> oldNodePrev = oldNode.prev;
+        MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(oldNode.prev, value, oldNode);
+        oldNode.prev = newNode;
+        if (oldNodePrev == null) {
+            head = newNode;
+        } else {
+            oldNodePrev.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            this.add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        MyLinkedList.Node<T> node = findNodeByIndex(index);
+        T oldVal = node.item;
+        node.item = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(findNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        while (node != null) {
+            if (object == node.item
+                    || (object != null && object.equals(node.item))) {
+                T deletedItem = unlink(node);
+                boolean result = ((deletedItem == object)
+                        || (object.equals(deletedItem)));
+                return result;
```

Comment:
```suggestion
                unlink(node);
                return true;
```
по факту, коли ми вже зайшли у if, то знаємо, що object це наш node.item, тому повторна перевірка зайва